app/src/main/java/com/android/shelfLife/model/newCreationScreen/CreationScreenViewModel.kt

```diff
@@ -0,0 +1,152 @@
+  val selectedHousehold = userRepository.selectedHousehold
+  val households = houseHoldRepository.households
+
+  var finishedLoading = MutableStateFlow(false)
+
+  // Fetch the list of members for the household to edit and set the email list
+  init {
+    viewModelScope.launch {
+      val members = houseHoldRepository.getHouseholdMembers(householdToEdit.value?.uid ?: "")
+      userRepository.getUserEmails(members) { uidToEmail ->
+        _emailList.value = uidToEmail.values.toSet()
+      }
+      finishedLoading.value = true
+    }
+  }
+
+  /**
+   * Sets the list of emails to the provided list. FOR TESTING PURPOSES ONLY.
+   *
+   * @param emails - The list of emails to set.
+   */
+  fun setEmails(emails: Set<String>) {
+    _emailList.value = emails
+  }
+
+  fun addEmail(email: String) {
+    _emailList.value += email
+  }
+
+  fun removeEmail(email: String) {
+    _emailList.value -= email
+  }
+
+  /**
+   * Gets the user IDs corresponding to a list of emails.
+   *
+   * @param emails - The list of emails to get user IDs for.
+   * @param callback - The callback to be invoked with the map of email to user ID.
+   */
+  fun getUserIdsByEmails(emails: Set<String>, callback: (Map<String, String>) -> Unit) {
+    userRepository.getUserIds(emails) { emailToUid -> callback(emailToUid) }
+  }
+
+  fun newHouseholdNameIsInvalid(householdName: String): Boolean {
+    return (householdName.isBlank() ||
+        houseHoldRepository.checkIfHouseholdNameExists(householdName) &&
+            (householdToEdit.value == null || householdName != householdToEdit.value!!.name))
+  }
+
+  /**
+   * Adds a new household to the repository and updates the household list.
+   *
+   * @param householdName - The name of the household to be added.
+   */
+  fun addNewHousehold(householdName: String, friendEmails: Set<String?> = emptySet()) {
+    val currentUser = FirebaseAuth.getInstance().currentUser
+    if (currentUser != null) {
+      val householdUid = houseHoldRepository.getNewUid()
+      var household = HouseHold(householdUid, householdName, listOf(currentUser.uid), emptyList())
+
+      if (friendEmails.isNotEmpty()) { // Corrected condition
+        userRepository.getUserIds(friendEmails) { emailToUserId ->
+          val emailsNotFound = friendEmails.filter { it !in emailToUserId.keys }
+          if (emailsNotFound.isNotEmpty()) {
+            Log.w("HouseholdViewModel", "Emails not found: $emailsNotFound")
+          }
+          viewModelScope.launch {
+            houseHoldRepository.addHousehold(household)
+            userRepository.addHouseholdUID(household.uid)
+            for (user in friendEmails.filter { it != currentUser.email }) {
+              invitationRepository.sendInvitation(household = household, invitedUserID = user!!)
+            }
+            if (selectedHousehold.value == null) {
+              Log.d("HouseholdViewModel", "Selected household is null")
+              userRepository.selectHousehold(
```

Comment:
Selecting households is done inside the user repository because everytime a houshold is selected, it has to be saved to the firebase. It could be a good idea to select the new household, but I'm not sure it is really necessary